### ISSUE TITLE
Use `crate_universe` instead of `cargo_raze`

### ISSUE THREAD

**Issue by aaronmondal** [allada/turbo-cache#115]
`cargo_raze` seems to be mostly abandoned in favor of `crate_universe`.

Moving from `cargo_raze` to `crate_universe` means essentially just changing build files like this:
```python
"//third-party:prost_types" -> "@crate_index//:prost-types"
```
The entire `third_party` directory is then superceded by a single `Cargo.Bazel.lock` file.

There are two ways `crate_universe` can be used:

1. Track dependencies in the WORKSPACE directly and delete `Cargo.toml`, i.e. something like
    ```python
    crates_repository(
        ...
        packages = {
            "somecrate": crate.spec(version = "1.2.3")
        },
    )
    ```
2. Keep the `Carto.toml` and generate dependencies from that, i.e. something like
    ```python
    crates_repository(
        ...
        manifests = ["@//:Cargo.toml"],
    )
    ```

@allada I already have an implementation of option 1 but I need to update it to the recently changed deps. I'll send a PR when It's ready :heart:

**Comment by allada** [comment]
Yeah the second option is more preferable. I don't like the idea of magic generated files much. Especially since `Cargo.toml` has a lot of magic to make things work.

**Comment by aaronmondal** [comment]
Addressed by #117
